cpp/src/io/utilities/config_utils.cpp

```diff
@@ -80,4 +87,36 @@ bool is_stable_enabled() { return is_all_enabled() or get_env_policy() == usage_
 
 }  // namespace nvcomp_integration
 
-}  // namespace cudf::io::detail
+inline std::mutex& host_mr_lock()
+{
+  static std::mutex map_lock;
+  return map_lock;
+}
+
+inline rmm::host_async_resource_ref default_pinned_mr()
+{
+  static rmm::mr::pinned_host_memory_resource default_mr{};
+  return default_mr;
+}
+
+inline auto& host_mr()
+{
+  static rmm::host_async_resource_ref host_mr = default_pinned_mr();
+  return host_mr;
+}
+
+}  // namespace detail
+
+void set_current_host_memory_resource(rmm::host_async_resource_ref mr)
+{
+  std::lock_guard<std::mutex> lock{detail::host_mr_lock()};
+  detail::host_mr() = mr;
+}
+
+rmm::host_async_resource_ref get_current_host_memory_resource()
```

Comment:
suggestion (non-blocking): You can drop the "current" if you want. I know this was based on `get_current_device_memory_resource()`, but there the "current" adjective describes the device, not the memory resource, i.e., "The memory resource on the _current device_." 

The closest equivalent for a host memory resource would be a NUMA node. If you wanted to be fancier, you could emulate the `per_device_memory_resource` logic and extend it to have `per_numa_node_memory_resource` as well as `get/set_nearest_numa_memory_resource()` that would query the NUMA node of the calling thread.

Or you could just skip the NUMA stuff all together.